### ISSUE TITLE
Using a field named query as GraphQL parameters breaks the generated code 

### ISSUE THREAD

**Issue by leandrolerena** [mirumee/ariadne#1166]
When using a query as parameter in a graphQL query, it gets overriden by the code generation query field. As a consequence, all graphQL queries which allow for a query parameter do not work. 

I propose to rename your variable to __query as of the GraphQL Spec (https://spec.graphql.org/October2021/): 

> Any [Name](https://spec.graphql.org/October2021/#Name) within a GraphQL type system must not start with two underscores "__" unless it is part of the [introspection system](https://spec.graphql.org/October2021/#sec-Introspection) as defined by this specification.

I think this would be a safe choice.

See an illustrative example below, where the argument gets directly overridden and as a consequence, the payload has syntax errors

```python
    async def devices(
        self,
        query: Union[Optional[str], UnsetType] = UNSET,
        page: Union[Optional[int], UnsetType] = UNSET,
        page_size: Union[Optional[int], UnsetType] = UNSET,
        id: Union[Optional[str], UnsetType] = UNSET,
        ordering: Union[Optional[str], UnsetType] = UNSET,
        scope: Union[Optional[Any], UnsetType] = UNSET,
        **kwargs: Any
    ) -> Devices:
        query = gql(
            """
```

**Comment by rafalp** [comment]
Hi!

I am afraid I don't understand what you mean here. Could you please provide a reproduction for Ariadne?

Thanks!

**Comment by leandrolerena** [comment]
Of course, sorry for being unclear.

The `devices` function is generated by ariadne. It is a GraphQL query generated from the schema. The query allows for 5 parameters:

1. query: Union[Optional[str], UnsetType] = UNSET,
2. page: Union[Optional[int], UnsetType] = UNSET,
3. page_size: Union[Optional[int], UnsetType] = UNSET,
4. id: Union[Optional[str], UnsetType] = UNSET,
5. ordering: Union[Optional[str], UnsetType] = UNSET,
6. scope: Union[Optional[Any], UnsetType] = UNSET, 

Which is perfect. However, in your function body, you define a field named `query`, and there comes the name-clash. The code-generator assumes that no parameter is called `query`, which is not an assumption backed by the GraphQL spec. 

You can see a scrreenshot from my editor here. The argument is greyed out, this means it is never used, as it gets directly overridden on line 735. Now instead of a valid query (like `hostname: a*`) the whole graphql query is passed, resulting in an invalid GraphQL request.

![image](https://github.com/mirumee/ariadne/assets/12090798/f67d6026-a591-4e3b-aed1-98581b54ea1a)

To sum up: The assumption of the code generator is that no query parameter is called `query`. If however such parameter exists, the generated code constructs an ill-formed payload which is then sent to the server. The payload would look something like:

![image](https://github.com/mirumee/ariadne/assets/12090798/50e9f702-3c80-44b9-9ebd-9d23d15488da)

as you can see, the query parameter contains the whole query again (as you can also observe in the generated code)

**Comment by leandrolerena** [comment]
My propsed correction would look like this (see comment in the issue description):

![image](https://github.com/mirumee/ariadne/assets/12090798/50499c10-9041-49ed-ada2-ffd2942ce487)

**Comment by rafalp** [comment]
Oh, so you mean Ariadne Codegen?

**Comment by leandrolerena** [comment]
Yes, exactly. I see now that the code for codegen is in another repo, I'm sorry for that. Should I recreate it there?
